### DISCUSSION TITLE
Is it possible to detect eslint errors before executing "next build", during "next dev"?

### DISCUSSION THREAD

**Discussion by raulesperante** [vercel/next.js#38345] [Help]
We always get this error during `next build`: 
`Build optimization failed: found page without a React Component as default export in...`
It would be useful to catch this error during development and not after running "next build".

I have done a search for similar discussions: https://github.com/vercel/next.js/issues/9904

**Comment by GabenGar** [comment]
Yes, by using eslint extension for your IDE.

**Comment by icyJoseph** [comment]
I don't think this is an ESLINT issue, is it? It is a build error. 

During development mode `next dev`, going to the page that's does not have a default export does crash though.

However, I don't think there's a lint rule for this, perhaps you want to write one? There's this one, https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/prefer-default-export.md, but it's not quite there... You kind want to make sure there's a default export, and not to prefer those over name exports.

Though, there's also `npx next lint` which can find other lint errors before you build :)

**Reply by raulesperante** [reply]
Thank you for your answer.
Yes, you are right. It is a build error.
I would like to detect this error while developing.
It is a good idea to write a new lint rule.

**Reply by icyJoseph** [reply]
I contradicted myself there. I mean it could be made into a ESLINT error, but it is not in the Next.js linting rules.

**Reply by nikhil-vinyl** [reply]
Thank you, I solved my errors using `npx next lint`

**Reply by GabenGar** [reply]
@nikhil-vinyl 
How did you do that when it was removed in next 16?
